src/libDirectoryService/ShardingPostProcessing.cpp

```diff
@@ -251,11 +251,32 @@ bool DirectoryService::ProcessShardingConsensus(
     // So, ANNOUNCE should acquire a lock here
 
     lock_guard<mutex> g(m_mutexConsensus);
-
+#if 1 //clark
+    // Wait until in the case that primary sent announcement pretty early
+#else
     unsigned int sleep_time_while_waiting = 100;
     // Wait for a while in the case that primary sent announcement pretty early
+#endif
     if ((m_state == POW2_SUBMISSION) || (m_state == SHARDING_CONSENSUS_PREP))
     {
+#if 1 //clark
+        cv_shardingConsensus.notify_all();
+
+        std::unique_lock<std::mutex> cv_lk(m_MutexCVShardingConsensusObject);
+
+        if (cv_shardingConsensusObject.wait_for(
+                cv_lk, std::chrono::seconds(CONSENSUS_OBJECT_TIMEOUT))
+            == std::cv_status::timeout)
+        {
+            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
```

Comment:
Maybe we can change this to a warning. If it timeout, it means something wrong has happens

Reply:
Fixed in a549b5576bea69b5c54149ac68a26af5474e9123